hack/update-vendor.sh

```diff
@@ -233,7 +233,9 @@ for repo in $(kube::util::list_staging_repos); do
   (
     echo "=== tidying k8s.io/${repo}"
     cd "staging/src/k8s.io/${repo}"
+    sed -E -i 's/^go 1\.(.*)$/go 1.21.0/' go.mod
```

Comment:
How is this resolving to 1.23.0 ? Module deps pushing it up?

Do we want to keep 1.21 as the floor? When will we upgrade it?

Reply:
i force it to 1.21.0 (picked at random, this could have been 1.22.0 too!), when i run `go mod tidy` it then resolves to 1.23.0 because some package or other needs 1.23.0 as minimum.

(we can pick whatever old we want on this line, we don't really need to update the line)